### PR TITLE
fix(logout): fix spark.logout() to fail gracefully

### PR DESCRIPTION
**Pull request by aimeex3**
fixes https://github.com/ciscospark/spark-js-sdk/issues/220

### CODE REVIEW

File: src/spark-core.js [hunk 1]
```diff
@@ -144,9 +144,9 @@ var SparkCore = AmpersandState.extend({
     return this.device.remove()
       .catch(function logDeviceRemovalFailure(reason) {
         this.logger.warn(reason);
-      })
+      }.bind(this))
       .then(function clearEncryptStore() {
-        this.encryption.keystore.clear();
+        return this.encryption.keystore.clear();
```
Comment (aimeex3) [inline src/spark-core.js]: @ianwremmel, is encryption always defined in legacy sdk or should we also check `if (this.encryption)`

**Comment by arun3528** [comment]
nice work aimee

File: src/client/device/device.js [hunk 2]
```diff
@@ -309,7 +309,7 @@ var Device = SparkBase.extend({
     this.logger.info('device: unregister requested');
 
     if (!this.url) {
-      throw new Error('device is not registered');
+      return Promise.reject('device is not registered');
```
Comment (mattnorris) [inline src/client/device/device.js]: 👍

File: src/spark-core.js (hunk 1 above)
Comment (ianwremmel) [inline src/spark-core.js]: yep, it's always defined

**Pull request merged**